pkg/controller/podautoscaler/horizontal.go

```diff
@@ -271,15 +271,15 @@ func (a *HorizontalController) reconcileAutoscaler(hpa *autoscaling.HorizontalPo
 	rescaleReason := ""
 	timestamp := time.Now()
 
-	if currentReplicas > hpa.Spec.MaxReplicas {
+	if currentReplicas == 0 {
```

Comment:
Makes sense for the 0 check. And should we use status.replicas for all the other cases?